File: Easy/9. Palindrome Number/02_attempt.py
#================================
# Working process:
#   1. Turn the integer into string
#   2. Check if the string is identical to the reversed string
# Refinement: More simple and readable code
# TakeAway: Learning how to slice and reverse string
#================================


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if str(x) == str(x)[::-1]:
            return True
        else:
            return False

File: Easy/9. Palindrome Number/02_attempt.cpp
//================================
// Working process:
//   1. Iterate through the first half of the string representation of x.
//   2. Compare the characters at the current index and the corresponding index from the end of the string.
//   3. If they are not equal, return false.
//   4. If the loop completes without returning false, return true.
// Refinement: The code converts the integer x to a string once and stores it in a variable, which improves efficiency and readability.
// TakeAway: Learning to write code in C++ and using string manipulation to check for palindromes.
//================================


class Solution {
public:
    bool isPalindrome(int x) {
        std::string s = std::to_string(x);

        for(int i=0;i<s.size()/2;++i){
            if(s[i] != s[s.size()-1-i]){
                return false;
            }
        }

        return true;
    }
};